### PR TITLE
Fix incorrect generators in converted IMA policies

### PR DESCRIPTION
**Pull request by mbestavros**
This could cause issues with the new integer generator format.

Signed-off-by: Mark Bestavros <mbestavr@redhat.com>

### CODE REVIEW

File: keylime/cmd/convert_ima_policy.py [hunk 1]
```diff
@@ -59,7 +59,7 @@ def convert_legacy_allowlist(allowlist_path: str) -> PolicyDict:
 def _convert_json_allowlist(alist_json: PolicyDict) -> PolicyDict:
     ima_policy = copy.deepcopy(ima.EMPTY_IMA_POLICY)
     ima_policy["meta"]["timestamp"] = str(datetime.datetime.now())
-    ima_policy["meta"]["generator"] = "convert_ima_policy.py"
+    ima_policy["meta"]["generator"] = ima.IMA_POLICY_GENERATOR.LegacyAllowList
```
Comment (aplanas) [inline keylime/cmd/convert_ima_policy.py:62]: yeah thanks, see "2" or "3" is really confusing : )

**Comment by mbestavros** [comment]
Tagging @maugustosilva since you introduced the `IMA_POLICY_GENERATOR` class that this adds support for.

**Review by aplanas — APPROVED** [review]

**Comment by aplanas** [comment]
> Tagging @maugustosilva since you introduced the `IMA_POLICY_GENERATOR` class that this adds support for.

Can you also change test/test_ima_verification.py and fix test/test_policy_conversion.py?

**Review by aplanas — CHANGES_REQUESTED** [review]
test/test_ima_verification.py and test/test_policy_conversion.py reference string or numeric values

File: test/test_policy_conversion.py [hunk 2]
```diff
@@ -55,7 +55,7 @@ def test_allowlist_conversion(self):
             created_ima_policy["meta"]["version"], ima.IMA_POLICY_CURRENT_VERSION, "Metadata version is correct"
         )
         self.assertEqual(
-            created_ima_policy["meta"]["generator"], "convert_ima_policy.py", "Metadata generator is correct"
+            created_ima_policy["meta"]["generator"], ima.IMA_POLICY_GENERATOR.LegacyAllowList, "Metadata generator is correct"
```
Comment (aplanas) [inline test/test_policy_conversion.py:58]: too long: https://github.com/keylime/keylime/actions/runs/3594545548/jobs/6052938461

File: test/test_policy_conversion.py [hunk 3]
```diff
@@ -74,3 +74,6 @@ def test_allowlist_conversion(self):
             )
         self.assertIsNotNone(created_ima_policy["excludes"], "Created IMA policy has 'excludes' field")
         self.assertIn("/usr/bin/zmore", updated_ima_policy["excludes"], "Sample exclusion path is correct")
+        self.assertEqual(
+            updated_ima_policy["meta"]["generator"], ima.IMA_POLICY_GENERATOR.CompatibleAllowList, "Metadata generator is correct"
```
Comment (aplanas) [inline test/test_policy_conversion.py:78]: https://github.com/keylime/keylime/actions/runs/3594545548/jobs/6052938461

**Review by aplanas — CHANGES_REQUESTED** [review]
missing here too: https://github.com/keylime/keylime/blob/master/test/test_ima_verification.py#L281

**Review by THS-on — APPROVED** [review]

**Comment by maugustosilva** [comment]
Thanks @aplanas ... Looks good @mbestavros, LGTM

**Review by maugustosilva — APPROVED** [review]

**Pull request merged**